src/errors/validation_exception.rs

```diff
@@ -123,13 +123,21 @@ impl<'a> IntoPy<ValError<'a>> for ValidationError {
 
 #[pymethods]
 impl ValidationError {
-    #[new]
-    fn py_new(title: PyObject, line_errors: &PyList, error_mode: Option<&str>) -> PyResult<Self> {
-        Ok(Self {
-            line_errors: line_errors.iter().map(PyLineError::try_from).collect::<PyResult<_>>()?,
-            title,
-            error_mode: ErrorMode::try_from(error_mode)?,
-        })
+    #[staticmethod]
+    fn from_exception_data(
+        py: Python,
+        title: PyObject,
+        line_errors: &PyList,
+        error_mode: Option<&str>,
+    ) -> PyResult<Py<Self>> {
+        Py::new(
+            py,
```

Comment:
Do you need to change this? Can't we just return a `self` like we did before?

Reply:
No, PyO3 does special handling for `#[new]` where it essentially wraps/swaps the result into a `Py`.